package/magento-dns/composition.yaml

```diff
@@ -0,0 +1,96 @@
+  resources:
+    - name: dns-zone
+      base:
+        apiVersion: dns.gcp.upbound.io/v1beta1
+        kind: ManagedZone
+        spec:
+          forProvider:
+            dnsName: ""
+            visibility: public
+      patches:
+      - type: FromCompositeFieldPath
+        fromFieldPath: spec.hostname
+        transforms:
+        - type: string
+          string: 
+            type: Format
+            fmt: "%s."
+        toFieldPath: spec.forProvider.dnsName
+      - type: FromCompositeFieldPath
+        fromFieldPath: metadata.annotations['crossplane.io/external-name']
+        toFieldPath: metadata.annotations['crossplane.io/external-name']
+      - type: ToCompositeFieldPath
+        fromFieldPath: metadata.annotations['crossplane.io/external-name']
+        toFieldPath: "status.dnsZone.name"
+      - type: FromCompositeFieldPath
+        fromFieldPath: status.providers.gcp.name
+        toFieldPath: "spec.providerConfigRef.name"
+
+    - name: dns-record-set
+      base:
+        apiVersion: dns.gcp.upbound.io/v1beta1
+        kind: RecordSet
+        spec:
+          forProvider:
+            managedZoneRef:
+              name: ""
+            name: ""
+            rrdatas: []
+            ttl: 300
+            type: A
+      patches:
+      - type: FromCompositeFieldPath
+        fromFieldPath: status.dnsZone.name
+        toFieldPath: "spec.forProvider.managedZoneRef.name"
+      - type: FromCompositeFieldPath
+        fromFieldPath: spec.hostname
+        transforms:
+          - type: string
+            string:
+              type: Format
+              fmt: "%s."
+        toFieldPath: spec.forProvider.name
+      - type: FromCompositeFieldPath
+        fromFieldPath: spec.ingressIP
+        toFieldPath: spec.forProvider.rrdatas[0]
+      - type: FromCompositeFieldPath
+        fromFieldPath: status.providers.gcp.name
+        toFieldPath: "spec.providerConfigRef.name"
+
+
+    - name: provider-config-gcp
+      base:
+        apiVersion: gcp.upbound.io/v1beta1
+        kind: ProviderConfig
+        spec:
+          credentials:
+            secretRef:
+              key: creds
+              name: ""
+              namespace: default
+            source: Secret
+          projectID: ""
+      patches:
+      - type: FromCompositeFieldPath
+        fromFieldPath: "spec.providerConfigSecretRef.secret.name"
+        toFieldPath: "spec.credentials.secretRef.name"
+      - type: FromCompositeFieldPath
+        fromFieldPath: "spec.providerConfigSecretRef.secret.namespace"
+        toFieldPath: "spec.credentials.secretRef.namespace"
+      - type: FromCompositeFieldPath
+        fromFieldPath: "spec.projectId"
+        toFieldPath: "spec.projectID"
+      - type: FromCompositeFieldPath
+        fromFieldPath: metadata.annotations['crossplane.io/external-name']
+        toFieldPath: "metadata.name"
+      - type: ToCompositeFieldPath
+        fromFieldPath: "metadata.name"
+        toFieldPath: "status.providers.gcp.name"
```

Comment:
@ca7alindev Exclude this provider, shall no be a part of configuration, user shall provider it on creation.
Because usually user have single provider config configured for access GCP, but in your case will be many of same provider configs.